// 1. Промисы: сравните then и catch
// Ответ: Нет, в первом примере, если ошибка произойдёт в f1, то она будет обработана в catch
// а во втором примере, ошибка не будет обработанна

promise.then(f1).catch(f2);
Против:
promise.then(f1, f2);